### ISSUE TITLE
Restructuring causing merged style rules to be overwritten

### ISSUE THREAD

**Issue by xeonicca** [jakubpawlowicz/clean-css#830]
Using clean-css@3.4.20 with the following rules:
```css
.button--black {
  color: tomato;
  border-color: black;
  border-bottom-color: rgba(0, 0, 0, .2);
}

.button--blue {
  color: tomato;
  border-color: blue;
  border-bottom-color: rgba(0, 0, 0, .2);
}
```

will be minified into the following (unminified for readability):
```css
.button--black, .button--blue {
  color: tomato;
  border-bottom-color: rgba(0, 0, 0, .2);
}

.button--black {
  border-color: #000;
}

.button--blue {
  border-color: #00f;
}
```

See the border bottom color will be overwritten because the merged rule is moved aboved `border-color:#000`. While turning off restructuring the code works as intended, but is this considered a bug on clean-css?

**Comment by jakubpawlowicz** [comment]
It's a bug indeed! Fix is coming.

**Comment by jakubpawlowicz** [comment]
It's fixed in 3.4.21.
